README.md

```diff
@@ -1,46 +1,103 @@
-# SecretHub Cross Language Golang Client<sup>**experimental**</sup>
+# SecretHub-XGO Guide for C# integration
 
-`secrethub-xgo` wraps the `secrethub-go` client with `cgo` exported functions so it can be called from other languages, e.g. C, C#, Python, Ruby, NodeJS, and Java. 
+This is a guide containing the following:
+ - [Prerequisites](#prerequisites)
+ - [Usage](#usage)
+ - [Building from source](#building-from-source)
+ - [How to call library functions](#how-to-call-library-functions)
+ - [Resources](#resources)
+ - [Getting help](#getting-help)
 
-At the moment, we provide the library for C# usage. 
+## Prerequisites
 
 In order to make use of the package, you will need to have the following installed:
  - [.NET Core](https://docs.microsoft.com/en-gb/dotnet/core/install/)
  - [Golang](https://golang.org/doc/install)
 
-## Create and use the package
+## Usage
 
-First, you have to make the NuGet Package and use it in your project. To do so, Here are the steps:
-1. execute `make nupkg` from the Makefile
-2. Go to your .NET project and run the following command: `dotnet add package SecretHub -s <path_to_your_secrethub-xgo_repo>`.
-3. Since our library uses Json.Net, you also need to add this dependency to your project. To do so, run the following command: `dotnet add package Newtonsoft.Json`.
+To install SecretHub package from NuGet Gallery, run the following command in your project's direcotry: 
+```bash
+dotnet add package SecretHub
+```
+or you can go to the project's `.csproj` file and add the following line:
+```xml
+<PackageReference Include="SecretHub" Version="0.1.0" />
+```
 
-## How to call library functions
-
-- Read
-`SecretHub.SecretVersion secret = SecretHub.Client.Read("path_to_secret");`
-
-- ReadString
-`string secret = SecretHub.Client.ReadString("path_to_secret");`
-
-- Exists
-`bool isSecret = SecretHub.Client.Exists("path_to_secret");`
-
-- Write
-`SecretHub.Client.Write("path_to_secret", "secret_value");`
+## Building from source 
+1. Execute `make nupkg` from the Makefile
+2. Go to your .NET project direcotry and run the following command: `dotnet add package SecretHub -s <path_to_your_secrethub-xgo_repo>`.
 
-- Remove
-`SecretHub.Client.Remove("path_to_secret");`
-
-- Resolve
-`string secret = SecretHub.Client.Resolve("reference");`
-
-- ResolveEnv
-`Doctionary<string, string> resolvedEnv = SecretHub.Client.ResolveEnv(dictionaryToResolve);`
-
-## Example
-
-An example project is found in `azure-example`. There you can see the library in action.
+## How to call library functions
+Before doing any calls to the library, you need to create you SecretHub client. This is done in the following way:
+```csharp
+var client = new SecretHub.Client();
+``` 
+or 
+```csharp
+SecretHub.Client client = new SecretHub.Client();
+```
+
```

Comment:
It's really complete to add this, but the added information for the reader is minimal, so i think we can just as well leave it out.
```suggestion
```

Reply:
Then, shall I keep the declaration with `var` or the object name?

Reply:
My C# knowledge is a bit rusty, but it being a `SecretHub.Client` is implied by the `new SecretHub.Client()`, right? As in, `new` shows we call a constructor, thereby it follows that the resulting type must be `SecretHub.Client`.

If so, replacing `var` with `SecretHub.Client` does not add any new information, so we can - in my opinion - leave it at `var`.